### PR TITLE
implement reset button for translator

### LINKED ISSUE #36: [Translator] Add a reset button

### PR DESCRIPTION
**Pull request by nycex**
fixes #36

### CODE REVIEW

**Review by cyyynthia — APPROVED** [review]
Changes looks ok imo, i'll not merge rn tho since i'm unable to test it
EDIT: According to Discord some behaviour is buggy

File: src/Powercord/plugins/translate.js [hunk 1]
```diff
@@ -21,21 +21,40 @@ module.exports = class Translate extends Plugin {
...
 
+        const timestamp = message.querySelector('.pc-timestampCozy');
         await Promise.all([
           sleep(200),
           Promise.all(
             [ ...message.querySelectorAll('.pc-markup') ]
               .map(async (markup) => {
                 const { text, from } = await translate(markup.innerText, opts);
+                if (!timestamp.innerHTML.includes('Translated from')) {
+                  markup.dataset.original = markup.innerText;
```
Comment (cyyynthia) [inline src/Powercord/plugins/translate.js]: innerHTML

File: src/Powercord/plugins/translate.js [hunk 2]
```diff
@@ -57,7 +76,8 @@ module.exports = class Translate extends Plugin {
                 .map(from => ({
                   type: 'button',
                   name: translate.languages[from],
-                  onClick: () => setText({ to, from })
+                  onClick: () => setText({ to,
```
Comment (cyyynthia) [inline src/Powercord/plugins/translate.js]: may be single lined?

**Review by cyyynthia — CHANGES_REQUESTED** [review]

File: src/Powercord/plugins/translate.js [hunk 3]
```diff
@@ -21,21 +21,40 @@ module.exports = class Translate extends Plugin {
...
         if (!timestamp.innerHTML.includes('Translated from')) {
-          timestamp.innerHTML += ` - Translated from ${fromLang}`;
+          timestamp.innerHTML += ' - ';
+          const span = document.createElement('span');
+          span.style = 'cursor: pointer';
+          span.innerText = `Translated from ${fromLang}`;
+          span.addEventListener('click', () => {
+            message.style.transition = '0.2s';
+            message.style.opacity = 0;
+            sleep(200);
```
Comment (devoxin) [inline src/Powercord/plugins/translate.js]: This needs to be awaited otherwise it's effectively redundant.

**Comment by cyyynthia** [comment]
http://sharing-memes-in-the-eu.is-my-k.ink/805c101c46.png and await here

File: src/Powercord/plugins/translate.js (hunk 2 above)
Comment (nycex) [reply src/Powercord/plugins/translate.js]: just ran eslint over it, and eslint made it two lined

Comment (cyyynthia) [reply src/Powercord/plugins/translate.js]: oh, fine then

File: src/Powercord/plugins/translate.js [hunk 4]
```diff
@@ -21,21 +21,39 @@ module.exports = class Translate extends Plugin {
...
-          timestamp.innerHTML += ` - Translated from ${fromLang}`;
+          timestamp.innerHTML += ' - ';
+          const span = document.createElement('span');
+          span.style = 'cursor: pointer';
+          span.innerText = `Translated from ${fromLang}`;
+          span.addEventListener('click', () => {
+            message.style.transition = '0.2s';
+            message.style.opacity = 0;
+            [ ...message.querySelectorAll('.pc-markup') ]
+              .map(async (markup) => {
```
Comment (cyyynthia) [inline src/Powercord/plugins/translate.js]: async is useless here

**Review by cyyynthia — CHANGES_REQUESTED** [review]

**Review by cyyynthia — APPROVED** [review]

**Pull request merged**